WeakAurasOptions/OptionsFrames/TextEditor.lua

```diff
@@ -509,6 +507,208 @@ local function ConstructTextEditor(frame)
+      APISearchCTimer:Cancel()
+    end
+    APISearchCTimer = C_Timer.NewTimer(
+      APISearchTextChangeDelay,
+      function()
+        makeAPISearch(filterInput:GetText())
+      end
+    )
+  end)
+  filterInput:SetHeight(15)
+  filterInput:SetPoint("TOPLEFT", apiSearchFrame, "TOPLEFT", 17, -30)
+  filterInput:SetPoint("TOPRIGHT", apiSearchFrame, "TOPRIGHT", -10, -30)
+  filterInput:SetFont(STANDARD_TEXT_FONT, 10, "")
+
+  local apiSearchScrollContainer = AceGUI:Create("SimpleGroup")
+  apiSearchScrollContainer:SetFullWidth(true)
+  apiSearchScrollContainer:SetFullHeight(true)
+  apiSearchScrollContainer:SetLayout("Fill")
+  apiSearchScrollContainer.frame:SetParent(apiSearchFrame)
+  apiSearchScrollContainer.frame:SetPoint("TOPLEFT", apiSearchFrame, "TOPLEFT", 17, -50)
+  apiSearchScrollContainer.frame:SetPoint("BOTTOMRIGHT", apiSearchFrame, "BOTTOMRIGHT", -10, 10)
+
+  local apiSearchScroll = AceGUI:Create("ScrollFrame")
+  apiSearchScroll:SetLayout("List")
+  apiSearchScrollContainer:AddChild(apiSearchScroll)
+  apiSearchScroll:FixScroll(true)
+  apiSearchScroll.scrollframe:SetScript(
+    "OnScrollRangeChanged",
+    function(frame)
+      frame.obj:DoLayout()
+    end
+  )
+
+  local snippetOnClickCallback = function(self)
+    if self.isSystem then
+      filterInput:SetText(self.name)
+    else
+      self.editor.editBox:Insert(self.name)
+      self.editor:SetFocus()
+    end
+  end
+
+  local function loadBlizzardAPIDocumentation()
+    local apiAddonName = "Blizzard_APIDocumentation"
+    local _, loaded = C_AddOns.IsAddOnLoaded(apiAddonName)
+    if not loaded then
+      C_AddOns.LoadAddOn(apiAddonName)
+    end
+    if #APIDocumentation.systems == 0 then
+      APIDocumentation_LoadUI()
+    end
+  end
+
+  local function addLine(results, apiInfo)
+    local name
+    if apiInfo.Type == "System" then
+      name = apiInfo.Namespace
+    elseif apiInfo.Type == "Function" then
+      name = apiInfo:GetFullName()
+    elseif apiInfo.Type == "Event" then
+      name = apiInfo.LiteralName
+    end
+    table.insert(results, { name = name, apiInfo = apiInfo })
+  end
+
+  local function APIListSystems()
+    local results = {}
+    for i, systemInfo in ipairs(APIDocumentation.systems) do
+      if systemInfo.Namespace and #systemInfo.Functions > 0 then
+        addLine(results, systemInfo)
+      end
+    end
+    table.sort(results, function(a, b)
+      return a.name < b.name
+    end)
+    return results
+  end
+
+  local function APISearch(word)
+    local lowerWord = word:lower()
+    local results = {}
+
+    -- if search field is set to the name of namespace, show all functions
+    local foundSystem = false
+    for _, systemInfo in ipairs(APIDocumentation.systems) do
+      -- search for namespaceName or namespaceName.functionName
+      local nsName, rest = lowerWord:match("^([%w%_]+)(.*)")
+      if nsName and systemInfo.Namespace and systemInfo.Namespace:lower():match(nsName) then
+        foundSystem = true
```

Comment:
The search algorithm feels off in two ways for me:

* Searching for e.g. "Cursor", doesn't find "CursorHasXXX", because there's an system C_Cursor. 
* Searching for e.g. "C_A" freezes the client for a bit since all functions in all systems that match "C_A" are displayed.

Reply:
your comments should be solved